### PR TITLE
Refactored prepare_for_export calls to ensure input size of example i…

### PR DESCRIPTION
**Pull request by borisfom**
…s handled properly

See also https://jirasw.nvidia.com/browse/JARS-1682

Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

### CODE REVIEW

File: nemo/collections/asr/modules/rnnt.py [hunk 1]
```diff
@@ -162,22 +162,14 @@ def __init__(
...
 
         # state maintenance is unnecessary during training forward call
         # to get state, use .predict() method.
-        if self._rnnt_export:
-            add_sos = False
-        else:
-            add_sos = True
-
-        g, states = self.predict(y, state=states, add_sos=add_sos)  # (B, U, D)
+        g, states = self.predict(y, state=states, add_sos=self.training)  # (B, U, D)
```
Comment (titu1994) [inline nemo/collections/asr/modules/rnnt.py:172]: This is incorrect - even during eval mode, we need an additional sos token in the forward. Only during export it should be one token (whatever it may be).

**Review by titu1994 — CHANGES_REQUESTED** [review]
Overall looks ok - but the rnn prediction network forward change is incorrect. Once that's reverted, should be good to merge

File: nemo/collections/asr/modules/rnnt.py (hunk 1 above)
Comment (borisfom) [reply nemo/collections/asr/modules/rnnt.py:172]: Oh I got confused by the comment above. Will put that back. They really should have had it tri-state (train/eval/infer) rather than boolean.

**Review by titu1994 — APPROVED** [review]
LGTM

**Review by VahidooX — APPROVED** [review]

File: nemo/collections/asr/modules/conv_asr.py [hunk 2]
```diff
@@ -76,7 +76,6 @@ def _prepare_for_export(self, **kwargs):
 
             if isinstance(m, SqueezeExcite):
                 m.set_max_len(int(one_hour // stride))  # One hour divided by current stride level
-                m.forward = m.forward_for_export
```
Comment (titu1994) [inline nemo/collections/asr/modules/conv_asr.py:79]: With the recent fix for DDP, will this line change cause problems ?

**Review by titu1994 — APPROVED** [review]
Overall looks good.
